File: matriz_mult/matriz_mult.py
def multiplicar_matrizes(A, B):
    m = len(A)
    n = len(A[0])
    p = len(B[0])
    C = [[0] * p for _ in range(m)]

    for i in range(m):
        for j in range(p):
            for k in range(n):
                C[i][j] += A[i][k] * B[k][j]

    return C


A = [[1, 2],
     [3, 4]]

B = [[5, 6],
     [7, 8]]

resultado = multiplicar_matrizes(A, B)
print(resultado)

File: matriz_mult/matriz_mult.c
#include <stdio.h>

void multiplicarMatrizes(int m, int n, int p, int A[m][n], int B[n][p], int C[m][p])
{
    for (int i = 0; i < m; i++)
    {
        for (int j = 0; j < p; j++)
        {
            for (int k = 0; k < n; k++)
            {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }
}

int main()
{
    int A[2][2] = {{1, 2}, {3, 4}};
    int B[2][2] = {{5, 6}, {7, 8}};
    int C[2][2] = {{0, 0}, {0, 0}};

    multiplicarMatrizes(2, 2, 2, A, B, C);

    printf("Resultado:\n");
    for (int i = 0; i < 2; i++)
    {
        for (int j = 0; j < 2; j++)
        {
            printf("%d ", C[i][j]);
        }
        printf("\n");
    }

    return 0;
}